.github/workflows/check-mergability.yaml

```diff
@@ -0,0 +1,38 @@
+# Checks for merge conflicts when a pull request is opened or updated
+name: Check mergability onto main
+
+on:
+  pull_request:
+  workflow_dispatch:
+
+jobs:
+  merge_check:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0  # This is important to fetch all history
+
+      - name: Check for merge conflicts
+        run: |
+          # Fetch the main branch
+          git fetch origin main:main
+
+          # Try to merge main onto the current branch
+          # The --no-commit and --no-ff flags make sure we don't actually
+          # commit a merge or create a merge commit, respectively.
+          MERGE_OUTPUT=$(git merge --no-commit --no-ff main) || true
```

Comment:
It could be easier to check the status code of the merge:
```
git merge main || MERGE_CONFLICT=1
if [ -z "$MERGE_CONFLICT" ]; then
   # all good
...
```

`--no-ff` probably could be omitted.

Reply:
Actually yes, as you want to explicitly prevent the merge from actually happening. For fast forwards, you can't do this.

Reply:
Do we really care whether merge actually happens or not for this case?